tsconfig.json

```
@@ -0,0 +1,20 @@
+{
+  "include": ["src/js/**/*"],
+  "compilerOptions": {
+    // Tells TypeScript to read JS files, as
+    // normally they are ignored as source files
+    "allowJs": true,
+    // Generate d.ts files
+    "declaration": true,
+    // This compiler run should
+    // only output d.ts files
+    "emitDeclarationOnly": true,
+    // Types should go into this directory.
+    // Removing this would place the .d.ts files
+    // next to the .js files
+    "outDir": "dist/types",
+    // go to js file when using IDE functions like
+    // "Go to Definition" in VSCode
+    "declarationMap": true
```

I think we should also add the following options:
```suggestion
    "declarationMap": true,
    "skipLibCheck": true,
    "checkJs": true,
    "preserveWatchOutput": true
```

---

I have just added your suggestions. However, I set the `checkJs` option to false because it throws an ocean of errors while creating definitions.